pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -270,10 +277,17 @@ impl InMemoryLayer {
     pub fn put_value(&self, key: Key, lsn: Lsn, val: &Value) -> Result<()> {
         trace!("put_value key {} at {}/{}", key, self.timelineid, lsn);
         let mut inner = self.inner.write().unwrap();
-
         inner.assert_writeable();
 
-        let off = inner.file.write_blob(&Value::ser(val)?)?;
+        let off = {
+            SER_BUFFER.with(|x| -> Result<_> {
+                let mut buf = x.borrow_mut();
+                val.ser_into(&mut (*buf))?;
+                let off = inner.file.write_blob(&buf)?;
+                buf.clear();
```

Comment:
Does `ser_into` overwrite existing data in the buffer, or append? If the `write_blob` call fails, `buf.clear()` is not called, so on the next call there will already be some data in the buffer. Best to call `buf.clear()` before `ser_into`.